Src/Base/AMReX_MultiFabUtil.cpp

```diff
@@ -1227,157 +1227,244 @@ namespace amrex
                                    noexcept -> Real
                     {
-                        constexpr Real pi = Real(3.1415926535897932);
-                        Real ri = rlo + dx[0]*i;
-                        Real ro = ri + dx[0];
-                        return Real(4./3.)*pi*(ro-ri)*(ro*ro+ro*ri+ri*ri)
-                            * a[box_no](i,j,k,icomp);
+                        if (m[box_no](i,j,k)) {
+                            return Real(0.);
+                        } else {
+                            constexpr Real pi = Real(3.1415926535897932);
+                            Real ri = rlo + dx[0]*i;
+                            Real ro = ri + dx[0];
+                            return Real(4./3.)*pi*(ro-ri)*(ro*ro+ro*ri+ri*ri)
+                                * a[box_no](i,j,k,icomp);
+                        }
                     });
                 } else
 #elif (AMREX_SPACEDIM == 2)
-                if (geom[nlevels-1].IsRZ()) {
-                    const auto rlo = geom[nlevels-1].CellSize(0);
-                    reduce_op.eval(*mf.back(), IntVect(0), reduce_data,
+                if (geom[ilev].IsRZ()) {
+                    const auto rlo = geom[ilev].CellSize(0);
+                    reduce_op.eval(*mf[ilev], IntVect(0), reduce_data,
                     [=] AMREX_GPU_DEVICE (int box_no, int i, int j, int k)
                                    noexcept -> Real
                     {
-                        Real ri = rlo + dx[0]*i;
-                        Real ro = ri + dx[0];
-                        constexpr Real pi = Real(3.1415926535897932);
-                        return pi*dx[1]*dx[0]*(ro+ri)
-                            * a[box_no](i,j,k,icomp);
+                        if (m[box_no](i,j,k)) {
+                            return Real(0.);
+                        } else {
+                            Real ri = rlo + dx[0]*i;
+                            Real ro = ri + dx[0];
+                            constexpr Real pi = Real(3.1415926535897932);
+                            return pi*dx[1]*dx[0]*(ro+ri)
+                                * a[box_no](i,j,k,icomp);
+                        }
                     });
                 } else
 #endif
                 {
-                    reduce_op.eval(*mf.back(), IntVect(0), reduce_data,
-                    [=] AMREX_GPU_DEVICE (int box_no, int i, int j, int k) noexcept
+                    reduce_op.eval(*mf[ilev], IntVect(0), reduce_data,
+                    [=] AMREX_GPU_DEVICE (int box_no, int i, int j, int k)
+                                   noexcept -> Real
                     {
-                        return dv*a[box_no](i,j,k,icomp);
+                        return m[box_no](i,j,k) ? Real(0.)
+                            : dv*a[box_no](i,j,k,icomp);
                     });
                 }
             }
+            Gpu::streamSynchronize();
+        }
+
+        auto const& a = mf.back()->const_arrays();
```

Comment:
Does it make sense to use `nlevels-1` here (and the following `reduce_op.eval` calls)? This is primarily a readability question (assuming I understood the length of mf)
```suggestion
        auto const& a = mf[nlevels-1]->const_arrays();
```

Reply:
Sorry.  Too Late.  I enabled auto-merge.   Nevertheless, one could argue that `mf.back()` is better because one is less likely to make a mistake such as `mf[nlevels]`.